cadence/src/sinks/unix.rs

```diff
@@ -218,6 +247,13 @@ impl MetricSink for BufferedUnixMetricSink {
         let mut writer = self.buffer.lock().unwrap();
         writer.flush()
     }
+
+    fn stats(&self) -> SinkStats {
+        SinkStats {
+            bytes_sent: self.bytes_sent.swap(0, Ordering::Relaxed),
```

Comment:
Resetting the counters when they're read seems very odd to me. What's the motivation for that? I'm used to the Prometheus world where counters are never reset (except on application restart).

Reply:
On my side I'm reporting it to DataDog continuously as counter so it makes more sense to reset in order to report bytes / packets sent since previous flush. I can for sure avoid resetting but not sure if this won't many users to some logic to do that anyway on their side.

Reply:
One alternative is to maybe impl operator traits to do easily subtraction - https://github.com/pretzelhammer/rust-blog/blob/master/posts/tour-of-rusts-standard-library-traits.md#operator-traits so I could use it on my side. Let me know what do you think.

Reply:
I'd rather not reset or implement any mathematical traits and let consumers of this handle computing the delta themselves.

Reply:
Because of orphan rule users won't be able to add e.g. `Sub` impl for `SinkStats` so e.g. in my case I would need to write code to calculate deltas. This is why I think that if we go with no-reset option then might be good to not make it unnecessary complex for users needing deltas instead.

Reply:
I'd be OK with something with the semantics of [`saturating_sub`](https://doc.rust-lang.org/std/primitive.u32.html#method.saturating_sub) but I don't want to add a `Sub` implementation and would rather let people manually write the code to do this if needed.